StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1504AllAccessorsMustBeSingleLineOrMultiLine.cs

```diff
@@ -96,35 +96,46 @@ private void HandlePropertyDeclaration(SyntaxNodeAnalysisContext context)
 
             if (property.AccessorList == null
                 || property.AccessorList.IsMissing
-                || property.AccessorList.Accessors.Count != 2)
+                || property.AccessorList.Accessors.Count < 2)
             {
                 return;
             }
 
-            if (!this.AreAllAccessorsInMultiLines(property.AccessorList.Accessors)
-               && !this.AreAllAccessorsInOneLine(property.AccessorList.Accessors))
+            var accessorsAnalyzeResult = this.AnalyzeAccessorsLineSpans(property.AccessorList.Accessors);
+            if (accessorsAnalyzeResult.SingleLineAccessors.Count == 1
+                && accessorsAnalyzeResult.MultiLineAccessors.Count == 1)
             {
-                context.ReportDiagnostic(Diagnostic.Create(Descriptor, property.AccessorList.Accessors[1].GetLocation()));
+                context.ReportDiagnostic(Diagnostic.Create(Descriptor, accessorsAnalyzeResult.SingleLineAccessors.Single().Keyword.GetLocation()));
```

Comment:
:exclamation: Use `First()` instead of `Single()` to guarantee we won't get an exception if there's ever a case where 3 or more accessors could exist (with 2 or more single-line, and 1 or more multi-line).